src/Uno.Toolkit.UI/Behaviors/VisualStateManagerExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using Uno.Extensions;
+using Uno.Logging;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Media;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+#endif
+
+namespace Uno.Toolkit.UI;
+
+/// <summary>
+/// Provides a way of manipulating the visual states of <see cref="Control"/> with attached property.
+/// </summary>
+/// <remarks>
+/// <seealso cref="VisualStateManager.GoToState"/> is typically used with <see cref="Control"/>
+/// where you would set <see cref="VisualStateManager.VisualStateGroupsProperty"/> on the root element of the ControlTemplate.
+/// Because this class directly calls that method, it means that if you are setting <see cref="OverrideStatesProperty"/> on an element,
+/// the <see cref="VisualStateManager.VisualStateGroupsProperty"/> should not be set on the very same element, but its first child.
+/// </remarks>
+public static class VisualStateManagerExtensions
+{
+	private static readonly ILogger Logger = typeof(VisualStateManagerExtensions).Log();
+
+	#region DependencyProperty: OverrideStates
+
+	/// <summary>
+	/// Identifies the OverrideStates dependency property.
+	/// </summary>
+	public static DependencyProperty OverrideStatesProperty { get; } = DependencyProperty.RegisterAttached(
+		"OverrideStates",
+		typeof(string),
+		typeof(VisualStateManagerExtensions),
+		new PropertyMetadata(default(string), OnOverrideStatesChanged));
+
+	/// <summary>
+	/// Sets the visual states of the control.
+	/// </summary>
+	/// <param name="obj"></param>
+	/// <param name="value">A space, comma or semi-colon separated list of visual state names</param>
+	public static void SetOverrideStates(Control obj, string value) => obj.SetValue(OverrideStatesProperty, value);
+
+	#endregion
+
+	private static void OnOverrideStatesChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+	{
+		if (sender is Control control && e.NewValue is string { Length: >0 } value)
+		{
+			var states = Regex.Split(value, "[,; ] *")
+				.Where(x => !string.IsNullOrEmpty(x))
+				.ToArray();
```

Comment:
Not sure if the Regex here will have a negative performance impact.

cc @jeromelaban